packages/http/src/router/matchBaseUrl.ts

```diff
@@ -1,23 +1,55 @@
+import * as E from 'fp-ts/lib/Either';
+import * as O from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+
 import { INodeVariable, IServer } from '@stoplight/types';
 import { MatchType } from './types';
 
 const variableRegexp = /{(.*?)}/g;
 
-export function matchBaseUrl(server: IServer, baseUrl: string): MatchType {
-  const templateMatchResult = matchRequestUrlToTemplateUrl(baseUrl, server.url, server.variables);
+export function matchBaseUrl(server: IServer, baseUrl: string) {
+  return pipe(
+    convertTemplateToRegExp(server.url, server.variables),
+    E.map(regex =>
+      pipe(
+        O.fromNullable(regex.exec(baseUrl)),
+        O.fold(
+          () => MatchType.NOMATCH,
+          matches => (matches.length > 1 ? MatchType.TEMPLATED : MatchType.CONCRETE)
+        )
+      )
+    ),
+    E.fold(
+      e => {
+        throw e;
+      },
+      matchType => matchType
+    )
```

Comment:
Don't do this — the idea of the monad is to propagate it back as it is unless you really need to handle the error. Just return the either and then we can use the `chain` method to use it back

Reply:
Got you, see comment above.